hamcrest/src/main/java/org/hamcrest/BaseDescription.java

```diff
@@ -29,6 +31,19 @@ public Description appendDescriptionOf(SelfDescribing value) {
     public Description appendValue(Object value) {
         if (value == null) {
             append("null");
+        } else if (value instanceof SelfDescribing){
```

Comment:
There is a fundamental problem with this approach: normally hamcrest would be a test dependency.